pennylane/operation.py

```diff
@@ -341,6 +360,21 @@ def check_domain(self, p, flattened=False):
             raise ValueError('{}: Unknown parameter domain \'{}\'.'.format(self.name, self.par_domain))
         return p
 
+    @property
+    def wires(self):
+        """Current wire values.
+
+        Fixed wires are returned as is, free wires represented by
+        :class:`~.variable.Variable` instances are replaced by their
+        current numerical value.
+
+        Returns:
+            list[int]: wire values
+        """
+        w = [i.val if isinstance(i, Variable) else i for i in self._wires]
+        self.check_wires(w)
+        return w
+
```

Comment:
Since wires can now be Variables, they need to be dynamically queried, instead of stored in a static attribute.